File: LaborAid/backend/app/schemas/enterprise.py
from pydantic import BaseModel, Field


class EnterpriseCompanyOut(BaseModel):
    id: str | None = None
    name: str
    credit_code: str | None = None
    reg_status: str | None = None
    legal_person: str | None = None
    reg_capital: str | None = None
    establish_time: str | None = None
    address: str | None = None
    business_scope: str | None = None
    company_type: str | None = None
    phone: str | None = None
    email: str | None = None
    website: str | None = None
    profile_url: str | None = None
    source: str = "qichacha"
    belong_org: str | None = None
    insured_count: str | None = None
    person_scope: str | None = None


class EnterpriseRiskSummaryOut(BaseModel):
    penalty_count: int = 0
    exception_count: int = 0
    shixin_count: int = 0
    zhixing_count: int = 0
    m_pledge_count: int = 0
    pledge_count: int = 0
    spot_check_count: int = 0
    has_liquidation: bool = False
    has_risk: bool = False


class EnterpriseRiskItemsOut(BaseModel):
    penalties: list[dict] = Field(default_factory=list)
    exceptions: list[dict] = Field(default_factory=list)
    shixin_items: list[dict] = Field(default_factory=list)
    zhixing_items: list[dict] = Field(default_factory=list)


class EnterpriseScanOut(BaseModel):
    search_key: str
    company: EnterpriseCompanyOut
    risk_summary: EnterpriseRiskSummaryOut
    risks: EnterpriseRiskItemsOut
    external_search_url: str | None = None
    disclaimer: str = "数据来源于企查查公开信息，仅供参考，请以市场监管部门及司法机关官方登记为准。"


class EnterpriseStatusOut(BaseModel):
    configured: bool
    provider: str = "qichacha"
    api_code: str = "736"
    message: str

File: LaborAid/frontend/src/lib/api/enterprise.ts
import { apiClient, TIMEOUT } from './client';

export interface EnterpriseCompany {
  id: string | null;
  name: string;
  credit_code: string | null;
  reg_status: string | null;
  legal_person: string | null;
  reg_capital: string | null;
  establish_time: string | null;
  address: string | null;
  business_scope: string | null;
  company_type: string | null;
  phone: string | null;
  email: string | null;
  website: string | null;
  profile_url: string | null;
  source: string;
  belong_org?: string | null;
  insured_count?: string | null;
  person_scope?: string | null;
}

export interface EnterpriseRiskSummary {
  penalty_count: number;
  exception_count: number;
  shixin_count: number;
  zhixing_count: number;
  m_pledge_count: number;
  pledge_count: number;
  spot_check_count: number;
  has_liquidation: boolean;
  has_risk: boolean;
}

export interface EnterpriseRiskItems {
  penalties: Record<string, unknown>[];
  exceptions: Record<string, unknown>[];
  shixin_items: Record<string, unknown>[];
  zhixing_items: Record<string, unknown>[];
}

export interface EnterpriseScanResult {
  search_key: string;
  company: EnterpriseCompany;
  risk_summary: EnterpriseRiskSummary;
  risks: EnterpriseRiskItems;
  external_search_url: string | null;
  disclaimer: string;
}

export interface EnterpriseStatus {
  configured: boolean;
  provider: string;
  api_code: string;
  message: string;
}

export const enterpriseApi = {
  status: async (): Promise<EnterpriseStatus> => {
    const res = await apiClient.get('/enterprise/status', { timeout: TIMEOUT.short });
    return res.data;
  },

  scan: async (searchKey: string): Promise<EnterpriseScanResult> => {
    const res = await apiClient.get('/enterprise/scan', {
      params: { search_key: searchKey },
      timeout: TIMEOUT.medium,
    });
    return res.data;
  },
};
